src/Models/Constants/Interfaces.ts

```diff
@@ -939,8 +940,13 @@ export interface IBlobServiceCorsRule {
     MaxAgeInSeconds: number;
 }
 
-export interface IDataHistoryWidgetTimeSeriesTwin {
+export interface IDataHistoryTimeSeriesTwin {
     label?: string;
     twinId: string;
     twinPropertyName: string;
+    twinPropertyType: PropertyValueType;
+    chartProps?: {
+        color?: string;
```

Comment:
no, these individual chart related props should be separately controlled and all optional. chartProps itself is also optional if you want to specify a property in it for the series. does it make sense?

Reply:
ok. I'm fine with the chartProps being optional, that makes sense that you wouldn't always want them, I just figured that if you were going to provide that object, you would want to provide all of the children since you're now saying you want to configure the chart. I'm generally hesitant to use a lot of optional props since it's not as clear to the consumer when they are actually required and when they aren't (and what happens when they are/aren't), so I tend to be fairly critical of them when I see them.

Reply:
I see, but most of the time chartProps are like optional/customizable chart options that you want or don't want to set. It is like optional settings for user's time series. On the contrary, I think they should be all optional in our case since we already have a default way of rendering them in the chart. If a user want to override or specify anything, they will use this chartProps as entry point for those parameters to set. I prefer to keep it as optional here.